lua/lvim/core/terminal.lua

```diff
@@ -77,23 +78,60 @@ local function get_dynamic_terminal_size(direction, size)
   end
 end
 
+local function term_toggle(term_opts)
+  local Terminal = require("toggleterm.terminal").Terminal
+  local term = Terminal:new(term_opts)
+  term:toggle(term_opts.size, term_opts.direction)
+end
+
+local function add_term_keymap(term_opts)
+  local binary = term_opts.cmd:match "(%S+)"
+  if vim.fn.executable(binary) ~= 1 then
+    Log:debug("Skipping configuring executable " .. binary .. ". Please make sure it is installed properly.")
+    return
+  end
+
+  local modes = { "n" }
+  if not term_opts.keymap:find "<leader>" then
+    table.insert(modes, "t")
+  end
+  vim.keymap.set(modes, term_opts.keymap, function()
+    term_toggle(term_opts)
+  end, { desc = term_opts.desc, noremap = true, silent = true })
+end
+
+--- Setup the terminal cmds
 M.init = function()
-  for i, exec in pairs(lvim.builtin.terminal.execs) do
-    local direction = exec[4] or lvim.builtin.terminal.direction
+  for i, term_opts in ipairs(lvim.builtin.terminal.terminals) do
+    -- size == 1 is a special case for full screen
+    if term_opts.size == 1 then
+      term_opts.direction = "float"
+      term_opts.float_opts = {
```

Comment:
This is actually pointless because it's handling `size = 1` which is a special case for fullscreen.
Nothing prevents you from setting `float_opts` and `direction = float` normally